Travis appraisal

Anbranin opened the pull request:
I mean don't merge this, but I could use some input. Please for the love of God help me.

Anbranin committed:
adding different rubies to travis

Anbranin committed:
adding scaffolding for appraisals

Anbranin committed:
add lockfiles to gitignore

Anbranin committed:
give travis gemfiles to run

Anbranin committed:
note to self

Anbranin committed:
added dummy apps and env variables for travis

ennukee commented:
😱

Anbranin committed:
trigger different ENV variable for each build matrix

Anbranin committed:
use custom script. exclude some jobs.

Anbranin committed:
add custom script

Anbranin committed:
exclude more jobs

Anbranin committed:
use the actual file name

Anbranin committed:
exclude more jobs. chmod script

Anbranin committed:
specify earlier webmock version

Anbranin committed:
do not need ruby-head, gorramit

Anbranin committed:
I thought the env variable test being set would do this but no?

Anbranin committed:
what even

Anbranin committed:
use bash script instead

Anbranin committed:
maybe the migrate and create need to be on separate lines

Anbranin committed:
testing script execution

Anbranin committed:
access script differently?

Anbranin commented:
Yeah I realize that now. I also realize I should have realized it was obvious that I was accessing the file correctly given that a 'permission denied' error occurred earlier.
I cannot tell if Travis is actually running the script, though, since I see no output. Am I supposed to? I made, as you can see, a test script that just echos. If it's getting run, should I see it in the build log, or is it normal that I don't?
I've read the entire first page of several Google searches and it appears I'm doing this correctly.
If I am and the scripts are indeed just getting silently run, then the problem lies in, perhaps, the placement of the script, or the migration not getting run correctly or in the right spot, or any number of things.
It's just frustrating running a migration and then being told I need to run a migration.

Anbranin committed:
HOW ABOUT THIS YOU FKER

Anbranin committed:
see if migrations run if not using script

Anbranin committed:
one line script work?

Anbranin committed:
use correct file name

Anbranin committed:
remove the other line

Anbranin committed:
what if it is the line at the top

Anbranin committed:
what if I put it here

werebus commented:
So 3 things to mention, as there appears to be a bit on confusion here.

1.  You don't need to change the file mode (`chmod`) every Travis run.  The script can just *be* executable in the git repository.  Git tracks file permissions too, so just chmod it on *your* computer and commit that.
2.  You have to *actually* execute the script.  Something like
    ```yaml
    before_script:
      - scripts/test.sh
    ```
3.  You shouldn't have both a set of environment variables *and* a set of gemfiles.  As it stands right now, your matrix is 2×3×3 (18 jobs before exclusions) and includes, for example, this one:
    ```json
    {
      "language": "ruby",
      "rvm": "2.3.1",
      "env": "RAILS_VERSION=4",
      "gemfile": "gemfiles/rails_5.gemfile",
    }
    ```

werebus commented:
Oh, and as far as whether to place your stuff in `script` or `before_script`:

`before_script` is part of the setup - it's stuff that *needs* to happen before we can even begin to test our software. If anything fails in that step, the build halts as an error becuase we can't run our tests.

`script` actually *is* the test step - it's stuff that determines whether our tests currently pass.  If anything fails in that step, the build continues (so we can see everything that's wrong), but the build will be a failure.

Anbranin committed:
gemfile.lock should not be checked into version control on gems or engines

Anbranin committed:
move gemfile bundling elsewhere. exclude certain jobs

Anbranin committed:
modify permissions on travis script.

remove unused scripts

Anbranin committed:
include different gemfile based on ENV variable

Anbranin committed:
remove duplicate references for gemspec

Anbranin committed:
change script to bundle different gemfiles

Anbranin committed:
remove case statement from Gemfile

Anbranin committed:
move script so appropriate bundling takes place

Anbranin committed:
put proper lines back in gemfiles

Anbranin committed:
try evaluating dif gemfile depending on ENV

Anbranin committed:
remove gemspec ref in gemfiles;

Anbranin committed:
expand to different file paths depending on env

Anbranin committed:
proper file paths

Anbranin committed:
remove appraisal

Anbranin committed:
migration should happen after migrations are created

Anbranin committed:
addressable breaks at 2.5 on ruby 1.9.3

Anbranin committed:
what if this worked

Anbranin committed:
this is probably not a thing

Anbranin committed:
this is probably not a thing

Anbranin committed:
script for special install

Anbranin committed:
add group to this one stupid gem

Anbranin committed:
move dumb gem to gemfile

Anbranin committed:
remove offending rubocop gem

Anbranin committed:
run proper install scripts

Anbranin committed:
proper gem name

Anbranin committed:
update syntax to run on ruby 1.9.3

Anbranin committed:
only pry-byebug in > rails 3

Anbranin committed:
ruby 1 syntax

Anbranin committed:
more syntax for old ruby

Anbranin committed:
I think these need to be on separate lines

Anbranin committed:
all the keys are strings now due to syntax change

Anbranin committed:
misspelled constant

werebus commented:
>  all the keys are strings now due to syntax change

This seems incorrect. That is, I don't think you should change the *behavior* of the application to accommodate an old version of ruby.  As a reminder, the following are all hashes with symbols for keys:

```ruby
# The *only* valid syntaxes in ruby 1.8.7
{ :key => 'value' }
{ :'key' => 'value' }
# Also valid in ruby 1.9+
{ key: 'value' }
# Also valid in ruby 2.2+
{ 'key': 'value' }
```

Anbranin committed:
keys are syntax again and valid in every ruby

Anbranin committed:
keys should in fact be strings---that is how they are expected to be in the controller

Anbranin committed:
use template instead

only template works for rails 3. Include Migrations class for rails
to get the migration_template method to work for rails 3.
Needed to implement own next_migration_number method also, again for rails 3.;

Anbranin committed:
whitelist assignable attributes

Anbranin committed:
add pry-debugger to rails-3

Anbranin committed:
readying for rails 3

Anbranin committed:
delete files that will be replaced,

or do not need to be in version control

Anbranin committed:
nah

ennukee commented on Gemfile:
```diff
@@ -10,6 +10,7 @@ gemspec
 # your gemspec. These might include edge Rails or gems from your path or
 # Git. Remember to move these dependencies to your gemspec before releasing
 # your gem to rubygems.org.
+# I DO NOT KNOW WHAT TO DO
```
:thinking:

dfaulken commented:
@Anbranin, can we close?

Anbranin commented:
UGh yes
